packages/@react-aria/menu/src/useSubmenuTrigger.ts

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaMenuItemProps} from './useMenuItem';
+import {AriaMenuOptions} from './useMenu';
+import type {AriaPopoverProps} from '@react-aria/overlays';
+import {FocusableElement, FocusStrategy, KeyboardEvent, PressEvent, Node as RSNode} from '@react-types/shared';
+import {RefObject, useCallback, useRef} from 'react';
+import type {SubmenuTriggerState} from '@react-stately/menu';
+import {useEffectEvent, useId, useLayoutEffect} from '@react-aria/utils';
+import {useLocale} from '@react-aria/i18n';
+import {useSafelyMouseToSubmenu} from './useSafelyMouseToSubmenu';
+
+export interface AriaSubmenuTriggerProps {
+  /** An object representing the submenu trigger menu item. Contains all the relevant information that makes up the menu item. */
+  node: RSNode<unknown>,
+  /** Whether the submenu trigger is disabled. */
+  isDisabled?: boolean,
+  /** Type of the submenu being rendered. */
+  submenuType?: 'dialog' | 'menu',
```

Comment:
def kinda weird, open to suggestions haha. Needed here so that contextual help dialogs don't get the same behavior as an actual submenu. Would we consider that Spectrum specific (didn't really feel like a uncommon use case though)?

 `suboverlayType` doesn't feel good

Reply:
just `type`? or is that not descriptive enough you think? We have something similar in `useMenuTrigger`